### PR TITLE
feat: Support integer on oracle ingestion/sink

### PR DESCRIPTION
**Pull request by karolisg**

### CODE REVIEW

File: dozer-ingestion/oracle/src/connector/mapping.rs [hunk 1]
```diff
@@ -51,7 +53,12 @@ fn map_data_type(
         match data_type {
             "VARCHAR2" => Ok(FieldType::String),
             "NVARCHAR2" => unimplemented!("convert NVARCHAR2 to String"),
-            "NUMBER" => Ok(FieldType::Decimal),
+            "INTEGER" => Ok(FieldType::I128),
+            "NUMBER" => match (precision, scale) {
+                (Some(precision), Some(0)) if precision >= 22 => Ok(FieldType::UInt),
```
Comment (Jesse-Bakker) [inline dozer-ingestion/oracle/src/connector/mapping.rs:58]: If precision >= 22, it won't fit in a UInt. Furthermore, `NUMBER` is signed.

File: dozer-ingestion/oracle/src/connector/replicate/log_miner/mapping.rs [hunk 2]
```diff
@@ -42,6 +42,12 @@ fn map_value(
         ))),
         (ParsedValue::String(string), FieldType::Decimal, _) => Ok(Field::Decimal(string.parse()?)),
         (ParsedValue::Number(number), FieldType::Decimal, _) => Ok(Field::Decimal(number)),
+        (ParsedValue::Number(number), FieldType::Int, _) => {
+            Ok(Field::Int(number.to_i64().unwrap()))
+        }
+        (ParsedValue::Number(number), FieldType::UInt, _) => {
+            Ok(Field::UInt(number.to_u64().unwrap()))
```
Comment (Jesse-Bakker) [inline dozer-ingestion/oracle/src/connector/replicate/log_miner/mapping.rs:49]: This unwrap might panic if the number is negative

File: dozer-sink-oracle/src/lib.rs [hunk 3]
```diff
@@ -246,10 +246,10 @@ impl OracleSinkFactory {
         for field in &schema.fields {
             let name = &field.name;
             let col_type = match field.typ {
-                dozer_types::types::FieldType::UInt => "NUMBER",
-                dozer_types::types::FieldType::U128 => unimplemented!(),
-                dozer_types::types::FieldType::Int => "NUMBER",
-                dozer_types::types::FieldType::I128 => unimplemented!(),
+                dozer_types::types::FieldType::UInt => "INTEGER",
+                dozer_types::types::FieldType::U128 => "INTEGER",
```
Comment (Jesse-Bakker) [inline dozer-sink-oracle/src/lib.rs:250]: Does `INTEGER` have a large enough precision? I thought it had precision `22`, which is only enough for about 70 bits, I believe

File: dozer-sink-oracle/src/lib.rs [hunk 4]
```diff
@@ -709,8 +714,14 @@ mod tests {
                 ON (D."id" = S."id" AND D."name" = S."name")
                 WHEN NOT MATCHED THEN INSERT (D."id", D."name", D."content", D."__txn_id", D."__txn_seq") VALUES (S."id", S."name", S."content", S."__txn_id", S."__txn_seq") WHERE S.DOZER_OPKIND = 0
                 WHEN MATCHED THEN UPDATE SET D."content" = S."content", D."__txn_id" = S."__txn_id", D."__txn_seq" = S."__txn_seq"
-                WHERE S.DOZER_OPKIND = 1 AND COALESCE(S."__txn_id" > D."__txn_id" OR (S."__txn_id" = D."__txn_id" AND S."__txn_seq" > D."__txn_seq"), TRUE) = TRUE
-                DELETE WHERE S.DOZER_OPKIND = 2 AND COALESCE(S."__txn_id" > D."__txn_id" OR (S."__txn_id" = D."__txn_id" AND S."__txn_seq" > D."__txn_seq"), TRUE) = TRUE
+                WHERE S.DOZER_OPKIND = 1 AND (S."__txn_id" IS NULL
```
Comment (Jesse-Bakker) [inline dozer-sink-oracle/src/lib.rs:717]: If `S.__txn_id IS NULL`, but `D.__txn_id IS NOT NULL`, we're in a weird state. That means that this was first replicated with a source that does send an OpIdentifier for records, and then for a source that doesn't.

File: dozer-sink-oracle/src/lib.rs (hunk 3 above)
Comment (karolisg) [reply dozer-sink-oracle/src/lib.rs:250]: From my understanding, oracle can guarantee precision up to 38.

File: dozer-sink-oracle/src/lib.rs (hunk 4 above)
Comment (karolisg) [reply dozer-sink-oracle/src/lib.rs:717]: Yeah, but not sure how to handle this

File: dozer-ingestion/oracle/src/connector/mapping.rs [hunk 5]
```diff
@@ -51,7 +53,12 @@ fn map_data_type(
         match data_type {
             "VARCHAR2" => Ok(FieldType::String),
             "NVARCHAR2" => unimplemented!("convert NVARCHAR2 to String"),
-            "NUMBER" => Ok(FieldType::Decimal),
+            "INTEGER" => Ok(FieldType::I128),
+            "NUMBER" => match (precision, scale) {
```
[the remark this answers is not in the document]
Comment (karolisg) [reply dozer-ingestion/oracle/src/connector/mapping.rs:57]: Right. Updated patterns.

File: dozer-sink-oracle/src/lib.rs (hunk 4 above)
Comment (karolisg) [reply dozer-sink-oracle/src/lib.rs]: Makes sense. Removed `S."__txn_id" IS NULL`

File: dozer-ingestion/oracle/src/connector/mapping.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (Jesse-Bakker) [reply dozer-ingestion/oracle/src/connector/mapping.rs:57]: `len(str(2**64)) == 20`
That means that if we want to guarantee that the number fits in a 64 bit integer, we should have precision <= 19. 
This is the opposite of the condition in the sink, where we're detecting if an integer fits into a destination column.

**Review by Jesse-Bakker — APPROVED** [review]
One small number change, otherwise LGTM

**Pull request merged**